src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,22 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private int height;
+    private int weight;
+    private String color;
+    private double maxSpeed;
+    private Engine engine;
 
+    public Car(int height, int weight, String color, double maxSpeed, Engine engine) {
+        this.height = height;
+        this.weight = weight;
+        this.color = color;
+        this.maxSpeed = maxSpeed;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        return new Car(this.height, this.weight, this.color, this.maxSpeed, this.engine);
```

Comment:
let's remove word `this` everywhere

and what about `engine`? how will we create clone of engine?